applications/views.py

```diff
@@ -0,0 +1,28 @@
+"""Views for bootcamp applications"""
+from django.http import Http404
+from rest_framework import viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+
+from applications.models import BootcampApplication
+from applications.serializers import BootcampApplicationDetailSerializer
+from main.permissions import UserIsOwnerPermission
+
+
+class BootcampApplicationViewset(viewsets.ReadOnlyModelViewSet):
+    """
+    View for fetching users' serialized bootcamp application(s)
+    """
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (IsAuthenticated, UserIsOwnerPermission,)
+    queryset = BootcampApplication.objects.prefetch_state_data()
+    owner_field = "user"
+
+    def get_serializer_class(self):
+        if self.action == "retrieve":
```

Comment:
I don't think this check is necessary here. Without the list view we only have `retrieve` as a valid action.